cassandra/src/test/scala/docs/scaladsl/CassandraFlowSpec.scala

```diff
@@ -0,0 +1,187 @@
+
+      val persons =
+        immutable.Seq(Person(12, "John", "London"), Person(43, "Umberto", "Roma"), Person(56, "James", "Chicago"))
+
+      val statementBinder: (Person, PreparedStatement) => BoundStatement =
+        (person, preparedStatement) => preparedStatement.bind(Int.box(person.id), person.name, person.city)
+
+      val written: Future[immutable.Seq[Person]] = Source(persons)
+        .via(
+          CassandraFlow.create(CassandraWriteSettings.defaults,
+                               s"INSERT INTO $table(id, name, city) VALUES (?, ?, ?)",
+                               statementBinder)
+        )
+        .runWith(Sink.seq)
+      // #prepared
+
+      written.futureValue must have size (persons.size)
+
+      val rows = CassandraSource(s"SELECT * FROM $table")
+        .map { row =>
+          Person(row.getInt("id"), row.getString("name"), row.getString("city"))
+        }
+        .runWith(Sink.seq)
+        .futureValue
+      rows must contain theSameElementsAs persons
+    }
+
+    "with context usage" in assertAllStagesStopped {
+      val table = createTableName()
+      withSchemaMetadataDisabled {
+        lifecycleSession.executeDDL(s"""
+                         |CREATE TABLE IF NOT EXISTS $table (
+                         |    id int PRIMARY KEY,
+                         |    name text,
+                         |    city text
+                         |);""".stripMargin)
+      }.futureValue mustBe Done
+
+      case class Person(id: Int, name: String, city: String)
+      case class AckHandle(id: Int) {
+        def ack(): Future[Done] = Future.successful(Done)
+      }
+      val persons =
+        immutable.Seq(Person(12, "John", "London") -> AckHandle(12),
+                      Person(43, "Umberto", "Roma") -> AckHandle(43),
+                      Person(56, "James", "Chicago") -> AckHandle(56))
+
+      // #withContext
+      val personsAndHandles: SourceWithContext[Person, AckHandle, NotUsed] = // ???
+        // #withContext
+        SourceWithContext.fromTuples(Source(persons))
+      // #withContext
+
+      val written: Future[Done] = personsAndHandles
+        .via(
+          CassandraFlow.withContext(
+            CassandraWriteSettings.defaults,
+            s"INSERT INTO $table(id, name, city) VALUES (?, ?, ?)",
+            (person, preparedStatement) => preparedStatement.bind(Int.box(person.id), person.name, person.city)
+          )
+        )
+        .asSource
+        .mapAsync(1) {
+          case (_, handle) => handle.ack()
+        }
+        .runWith(Sink.ignore)
+      // #withContext
+
+      written.futureValue mustBe Done
+
+      val rows = CassandraSource(s"SELECT * FROM $table")
+        .map { row =>
+          Person(row.getInt("id"), row.getString("name"), row.getString("city"))
+        }
+        .runWith(Sink.seq)
+        .futureValue
+      rows must contain theSameElementsAs persons.map(_._1)
+    }
+
+    "allow unlogged batches" in assertAllStagesStopped {
```

Comment:
Curious what makes this an "unlogged" batch?  According to the `CassandraFlow.createBatch` scaladocs:

```
   * Cassandra batches that share the same partition key will only
   * resolve to one write internally in Cassandra, boosting write performance.
```

Do the input `persons` share the same partition key?

Reply:
The test names wasn't updated.